packages/vite/src/node/optimizer/index.ts

```diff
@@ -761,7 +761,7 @@ async function prepareEsbuildOptimizerRun(
     ),
   }
 
-  const platform = environment.config.webCompatible ? 'browser' : 'node'
+  const platform = environment.config.consumer === 'client' ? 'browser' : 'node'
```

Comment:
I think this change is not correct, since there might be webCompatible servers such as Cloudflare's workerd

Reply:
linking for future reference: https://github.com/vitejs/vite/pull/18611